test/foundry/deploy/03_Erc20Enforcer.t.sol

```diff
@@ -0,0 +1,697 @@
+
+  function testBurn() public {
+    mintToDeployer();
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), zeroAddress, initialValue);
+    vm.prank(deployer);
+    holographERC20.burn(initialValue);
+  }
+
+  function testBurnFromNotApproveRevert() public {
+    vm.expectRevert("ERC20: amount exceeds allowance");
+    holographERC20.burnFrom(deployer, initialValue);
+  }
+
+  function testBurnFrom() public {
+    mintToDeployer();
+    approvalToAlice(initialValue);
+    vm.expectEmit(true, true, false, true);
+    emit Transfer(address(deployer), zeroAddress, initialValue);
+    vm.prank(alice);
+    holographERC20.burnFrom(deployer, initialValue);
+  }
+
+  /*
+   * ERC20 TEST
+   *    Permit
+   */
+
+  //TODO buildDomainSeperator
+
+  function testPermitZeroNounce() public {
+    assertEq(holographERC20.nonces(alice), 0);
+  }
+
+  function testPermitBadDeadLineRevert() public {
+    vm.expectRevert("ERC20: expired deadline");
+    holographERC20.permit(deployer, alice, initialValue, badDeadLine, uint8(0x00), zeroSignature, zeroSignature);
+  }
+
+  function testPermitEmptySignatureRevert() public {
+    console.log("block.timestamp", block.timestamp);
+    console.log("block.goodDeadLine", goodDeadLine);
+    console.log("block.badDeadLine", badDeadLine);
+    vm.expectRevert("ERC20: zero address signer");
+    holographERC20.permit(deployer, alice, initialValue, goodDeadLine, uint8(0x1b), zeroSignature, zeroSignature);
+  }
+
+  //TODO see, for me not work fine, 0x1b always rever for zerdo address
+  function testPermitZeroAddressSignatureRevert() public {
+    vm.expectRevert("ERC20: zero address signer");
+    holographERC20.permit(deployer, alice, initialValue, goodDeadLine, uint8(0x1b), signature, signature);
+  }
+
+  function testPermitInvalidSignatureV_ValueRevert() public {
+    vm.expectRevert("ERC20: invalid v-value");
+    holographERC20.permit(deployer, alice, initialValue, goodDeadLine, uint8(0x04), zeroSignature, zeroSignature);
+  }
+
+  //TODO not revert whit invalid signature
+  function testPermitInvalidSignatureRevert() public {
+    vm.skip(true);
+    vm.expectRevert("ERC20: invalid signature");
+    vm.prank(deployer);
+    holographERC20.permit(deployer, alice, initialValue, goodDeadLine, uint8(0x1b), signature2, signature3);
+  }
+
+  //TODO add sucess Permit
+
+  /*
+   * ERC20 TEST
+   *    Ownership tests
+   */
+
+  function testOwner() public {
+    assertEq(holographERC20.owner(), deployer);
+  }
+
+  function testIsOwner() public {
+    vm.prank(deployer);
+    assertEq(sampleERC20.isOwner(), true);
+  }
+
+  function testIsOwnerFalse() public {
+    vm.prank(alice);
+    assertEq(sampleERC20.isOwner(), false);
+  }
+
+  function testErc20GetOwnerProxy() public {
+    assertEq(holographERC20.getOwner(), address(admin));
+  }
+
+  function testErc20DeployerTransferOwnerRevert() public {
+    vm.expectRevert("HOLOGRAPH: owner only function");
+    holographERC20.setOwner(alice);
```

Comment:
suggestion: let's add a vm.prank(alice) here, to enforce the idea that the caller is not the owner

Reply:
ok

Reply:
done!